tests/MutationTest.php

```diff
@@ -0,0 +1,1682 @@
+                        'id_license' => 'VF',
+                        'age'        => 15,
+                    ],
+                    [
+                        'id_program' => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+                        'id_version' => '1.0',
+                        'id_license' => 'VF',
+                        'age'        => 15,
+                    ],
+                ],
+            ],
+        ];
+        $this->assertEquals($expectedMutationData, $mutation->jsonSerialize());
+    }
+
+    /**
+     * @test
+     */
+    public function whenFilteredItemsFromACollectionAreUpdated()
+    {
+        $program = new QueryItem([
+            'id_program'   => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+            'id_developer' => null,
+            'id_category'  => 'games',
+            'versions'     => new QueryCollection([
+                new QueryItem([
+                    'id_program' => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+                    'id_version' => '3.0',
+                    'id_license' => 'VF',
+                    'age'        => null,
+                ]),
+                new QueryItem([
+                    'id_program' => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+                    'id_version' => '2.0',
+                    'id_license' => 'VF',
+                    'age'        => 18,
+                ]),
+                new QueryItem([
+                    'id_program' => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+                    'id_version' => '1.0',
+                    'id_license' => 'VF',
+                    'age'        => null,
+                ]),
+            ]),
+        ]);
+
+        $mutation = new Mutation($this->configMock, $program);
+
+        $mutation->versions->upsert->filter(['age' => null])->set([
+            'id_license' => 'Fw',
+            'age'        => 15,
+        ]);
+
+        $expectedMutationData = [
+            'id_program'   => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+            'id_developer' => null,
+            'id_category'  => 'games',
+            'versions'     => [
+                'upsert' => [
+                    [
+                        'id_program' => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+                        'id_version' => '3.0',
+                        'id_license' => 'Fw',
+                        'age'        => 15,
+                    ],
+                    [
+                        'id_program' => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+                        'id_version' => '1.0',
+                        'id_license' => 'Fw',
+                        'age'        => 15,
+                    ],
+                ],
+            ],
+        ];
+        $this->assertEquals($expectedMutationData, $mutation->jsonSerialize());
+    }
+
+    /**
+     * @test
+     */
+    public function whenAnItemIsAddedAndAFilterIsAppliedToUpdateTheFilteredItemsAndFinallyAnotherItemIsAdded()
```

Comment:
My 2 cents: long test names are hard to read, we should reduce it to the minimum. If we are interested in more details of the content, we'll read the code

Reply:
AFAIK read the code is harder than read a method name